yasnippet.el

```diff
@@ -3910,40 +3910,67 @@ Meant to be called in a narrowed buffer, does various passes"
     (goto-char parse-start)
     (yas--indent snippet)))
 
+(defun yas--regexp-quote-flexspace (beg end)
+  "Return regexp match text from BEG to END allowing for whitespace changes."
+  (let ((regexps nil)
+        (pt beg))
+    (save-excursion
+      (goto-char pt)
+      (while (re-search-forward "[[:space:]]+" end t)
+        (let ((spaces (match-string-no-properties 0)))
+          (push (regexp-quote (buffer-substring-no-properties
+                               pt (match-beginning 0)))
+                regexps)
+          (setq pt (point))
+          ;; Prefer exact match, but allow some fuzzyness for spaces.
+          (push (concat "\\(?:" spaces "\\|[[:space:]]*\\)")
+                regexps)))
+      (push (regexp-quote (buffer-substring-no-properties
+                           pt end))
+            regexps))
+    (apply #'concat (nreverse regexps))))
+
 (defun yas--indent-region (from to snippet)
   "Indent the lines between FROM and TO with `indent-according-to-mode'.
 The SNIPPET's markers are preserved."
-  ;;; Apropos indenting problems....
```

Comment:
Looks great from afar :-), especially since you added texts, but are you sure you want to delete this explanation forever? It would be worth rewriting it explain the new implementation in my opinion...